[CI] CrossClusterAsyncSearchIT testCCSClusterDetailsWhereAllShardsSkippedInCanMatch failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4544 / windows-2016_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/ltn5jttwchx7u)
- [elasticsearch-pull-request #37735 / part-2](https://gradle-enterprise.elastic.co/s/jtdqh4l644x5g)
- [elasticsearch-pull-request #37687 / part-2](https://gradle-enterprise.elastic.co/s/y5px2fiz7obem)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:async-search:internalClusterTest" --tests "org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT.testCCSClusterDetailsWhereAllShardsSkippedInCanMatch" -Dtests.seed=61F31F8B253C9711 -Dtests.locale=ak-Latn-GH -Dtests.timezone=America/Adak -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCCSClusterDetailsWhereAllShardsSkippedInCanMatch),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 3 failures in test testCCSClusterDetailsWhereAllShardsSkippedInCanMatch (0.5% fail rate in 600 executions)
- [main] 2 failures in step part-2 (0.8% fail rate in 237 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.8% fail rate in 243 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

The failure seems to be: 
```

2> REPRODUCE WITH: ./gradlew ":x-pack:plugin:async-search:internalClusterTest" --tests "org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT.testCCSClusterDetailsWhereAllShardsSkippedInCanMatch" -Dtests.seed=1E450CED74B96307 -Dtests.locale=to-Latn-TO -Dtests.timezone=Indian/Mahe -Druntime.java=22 |  
-- | --
  | 2> java.lang.AssertionError: search cluster results should be marked as partial |  
  | at __randomizedtesting.SeedInfo.seed([1E450CED74B96307:D7FCF8BC0DE69943]:0) |  
  | at org.junit.Assert.fail(Assert.java:89) |  
  | at org.junit.Assert.assertTrue(Assert.java:42) |  
  | at org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT.testCCSClusterDetailsWhereAllShardsSkippedInCanMatch(CrossClusterAsyncSearchIT.java:291)

```
Does not reproduce locally :\

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -267,9 +267,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/115631
-- class: org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT
-  method: testCCSClusterDetailsWhereAllShardsSkippedInCanMatch
-  issue: https://github.com/elastic/elasticsearch/issues/115652
 - class: org.elasticsearch.index.get.GetResultTests
   method: testToAndFromXContent
   issue: https://github.com/elastic/elasticsearch/issues/115688

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java
@@ -287,7 +287,6 @@ public class CrossClusterAsyncSearchIT extends AbstractMultiClustersTestCase {
         try {
             responseId = response.getId();
             assertNotNull(response.getSearchResponse());
-            assertTrue(response.isRunning());
             SearchResponse.Clusters clusters = response.getSearchResponse().getClusters();
             assertThat(clusters.getTotal(), equalTo(2));
             if (dfs) {

